activerecord/lib/active_record/attribute_methods/dirty.rb

```diff
@@ -101,9 +101,6 @@ def attribute_changed_in_place?(attr_name)
       private
 
         def mutation_tracker
-          unless defined?(@mutation_tracker)
-            @mutation_tracker = nil
```

Comment:
Looks like it was to avoid test error https://github.com/rails/rails/commit/ede244e2367c7988fa02f5e3b956d88211bb5ade.

@sgrif do you remember?